Add doc for filter support for cli tools (#11012)
gravitational/teleport: docs/pages/access-controls/guides/impersonation.mdx

$ tctl auth sign --user=security-scanner --format=openssh --out=security-scanner --ttl=10h
```

> ### Filter fields explained

Reviewer: ```suggestion
### Filter fields
```
I think it's implied that this will be an explanation of the filter fields.

> Explanation of the fields used in this guide for the `where` conditions.

Reviewer: ```suggestion
Here is an explanation of the fields used in the `where` conditions within this guide.
```
Suggested making this a complete sentence to create a slightly more conversational tone.

| Field                                             | Description                                                    |
|---------------------------------------------------|----------------------------------------------------------------|

> | `user.spec.traits["<list name>"]`                 | the `list` of traits from a local or SSO user                  |

Reviewer: I don't see `list` used literally within a configuration resource, so "the `list`" might not be the clearest description to use here. Is there another way to phrase this that might be more recognizable? In other words, where would a user know how to find the value to use here?

Author: actually, the doc says that  `user traits typically come from an external identity provider`, so this tells me this value will rarely change, so instead of `<list name>` i put the actual value of what it will typically be

> | `impersonate_role.metadata.labels["<label key>"]` | the label `value` given label `key` from a role to impersonate |

Reviewer: ```suggestion
| `impersonate_role.metadata.labels["<label key>"]` | The label value given the label `key` from a role to impersonate |
```
Since `value` isn't a keyword used in the role resources above or in the table cell to the left of the explanation, I thought it would make sense to remove the code style.

> | `impersonate_user.metadata.labels["<label key>"]` | the label `value` given label `key` from a user to impersonate |

Reviewer: ```suggestion
| `impersonate_user.metadata.labels["<label key>"]` | The label value given the label `key` from a user to impersonate |
```
same re: `value`
